Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -0,0 +1,169 @@
+
+}
+
+class OfflineStoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
+
+    var serverIsDown: Bool = false
+    override var forceServerErrors: Bool { return self.serverIsDown }
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        await self.waitForPendingCustomerInfoRequests()
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+            try await self.ensureEntitlementMappingIsAvailable()
+        }
+    }
+
+    func testOfferingsAreCachedInMemory() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings) === onlineOfferings
+    }
+
+    func testOfferingsAreCachedOnDisk() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+        await self.resetPurchases()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings.response) == onlineOfferings.response
+
+        let offering = try XCTUnwrap(offlineOfferings.current)
+        expect(offering.availablePackages.count) == onlineOfferings.current?.availablePackages.count
+        expect(offering.monthly?.storeProduct.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testOfflineCustomerInfoWithNoPurchases() async throws {
+        Purchases.shared.invalidateCustomerInfoCache()
+
+        self.serverDown()
+
+        let info = try await Purchases.shared.customerInfo()
+        expect(info.entitlements.all).to(beEmpty())
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testOfflineCustomerInfoWithOnePurchase() async throws {
+        try await self.purchaseMonthlyProduct()
+
+        Purchases.shared.invalidateCustomerInfoCache()
+        self.serverDown()
+
+        let info = try await Purchases.shared.customerInfo()
+        expect(info.entitlements.all).toNot(beEmpty())
+        try await self.verifyEntitlementWentThrough(info)
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseWhileServerIsDownSucceedsButDoesNotFinishTransaction() async throws {
+        let logger = TestLogHandler()
+
+        self.serverDown()
+        try await self.purchaseMonthlyProduct()
+
+        logger.verifyMessageWasLogged(Strings.offlineEntitlements.computing_offline_customer_info, level: .info)
+        logger.verifyMessageWasNotLogged("Finishing transaction")
+    }
+
+    @available(iOS 15.2, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseWhileServerIsDownPostsReceiptAfterServerComesBack() async throws {
```

Comment:
I'm glad I'm running these in both SK1 and SK2. Turns out this is failing in SK1 only.

Reply:
For some reason, after restarting the SDK and recreating `SKPaymentQueue`, the SDK isn't getting notified about the transaction again, unlike with SK2. Need to figure that out.